src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2327,6 +2333,198 @@
+                        IDebugStackFrame2 frame = varScope.StackFrame;
+                        m_variableManager.TryGetProperty((frame, name), out property);
+                    }
+                    else if (variableObj is VariableEvaluationData varEvalData)
+                    {
+                        // We have a variable parent object.
+                        IDebugProperty2 parentProperty = varEvalData.DebugProperty;
+                        m_variableManager.TryGetProperty((variableReference, name), out property);
+                    }
+                }
+                else
+                {
+                    // We don't have a parent object. Default to using top stack frame
+                    if (m_frameHandles == null || !m_frameHandles.TryGetFirst(out IDebugStackFrame2 frame))
+                    {
+                        response.Description = string.Format(CultureInfo.CurrentCulture, AD7Resources.Error_DataBreakpointInfoFail, AD7Resources.Error_NoParentObject);
+                    }
+                    else
+                    {
+                        m_variableManager.TryGetProperty((frame, name), out property);
+                    }
+                }
+
+                // If we've found a valid child property to set the data breakpoint on, get the address/size and return the DataId.
+                if (property != null && property is IDebugProperty160 property160)
+                {
+                    hr = property160.GetDataBreakpointInfo160(out string address, out uint size, out string displayName, out errorMessage);
+                    eb.CheckHR(hr);
+                    if (!string.IsNullOrEmpty(errorMessage))
+                    {
+                        response.Description = string.Format(CultureInfo.CurrentCulture, AD7Resources.Error_DataBreakpointInfoFail, errorMessage);
+                    }
+                    else
+                    {
+                        // If we succeeded, return response that we can set a data bp.
+                        string sSize = size.ToString(CultureInfo.InvariantCulture);
+                        response.DataId = $"{address},{sSize}";
+                        response.Description = string.Format(CultureInfo.CurrentCulture, AD7Resources.DataBreakpointDisplayString, displayName, sSize);
+                        response.AccessTypes = new List<DataBreakpointAccessType>() { DataBreakpointAccessType.Write };
+                    }
+                }
+                else if (response.Description == null)
+                {
+                    response.Description = string.Format(CultureInfo.CurrentCulture, AD7Resources.Error_DataBreakpointInfoFail, AD7Resources.Error_ChildPropertyNotFound);
+                }
+            }
+            catch (Exception ex)
+            {
+                response.Description = string.Format(CultureInfo.CurrentCulture, AD7Resources.Error_DataBreakpointInfoFail, "");
```

Comment:
>response.Description [](http://example.com/codeflow?start=16&length=20)

Should this be `responder.SetError(...)` instead and move `responder.SetResponse` into the try block?

Reply:
Oh, I see.